commit 15f823b43ed3a525cc446d8b885d61f932891544
Author: Sage Weil <sweil@redhat.com>
Date:   Tue May 23 15:17:34 2017 -0400

    mgr/MgrStandby: reset subscriptions when we become non-active
    
    This is a goofy workaround that we're also doing in Mgr::init().  Someday
    we should come up with a more elegant solution.  In the meantime, this
    works just fine!
    
    Signed-off-by: Sage Weil <sage@redhat.com>

diff --git a/src/mgr/MgrStandby.cc b/src/mgr/MgrStandby.cc
index a6b76643b73..3d68cfe7f7e 100644
--- a/src/mgr/MgrStandby.cc
+++ b/src/mgr/MgrStandby.cc
@@ -251,6 +251,11 @@ void MgrStandby::handle_mgr_map(MMgrMap* mmap)
       derr << "I was active but no longer am" << dendl;
       active_mgr->shutdown();
       active_mgr.reset();
+
+      // FIXME: reset monc connection so that our old subscriptions go away
+      // and we stop getting MLog and MMgrDigest messages.  (We do something
+      // similar in Mgr::init().)
+      monc.reopen_session();
     }
   }
 
